data/Dockerfiles/netfilter/server.py

```diff
@@ -119,6 +127,19 @@ def ban(address):
   self_network = ipaddress.ip_network(address.decode('ascii'))
   if WHITELIST:
     for wl_key in WHITELIST:
+      if not is_ip_network(wl_key):
+        hostname = wl_key
+        try:
+          wl_key = socket.gethostbyname(hostname)

```

Comment:
You know how long this takes to timeout if the name is invalid? :)

Reply:
I don't think invalid names are a problem, the bigger problem is a slow responding dns.

Hm, what think you about a local cache? 
I looked into the Fail2ban source, they solve this issue with a cache ^^.

Maybe you have a suggestion?

Reply:
Timeout of 1 sec is ok?

Reply:
Oh, sorry, feel free to change it to whatever you think fits best!

Switching to Py3 would be great. No reason to stick with Py2. :)